### PR TITLE
Linear Pools: remove price rate cache

### PR DESCRIPTION
**Pull request by TomAFrench**
This PR simply removes the price rate cache from the Linear Pool and creates a virtual `_getWrappedTokenRate` function for derived contracts to specify. A `MockLinearPool` has been created with a constant rate to maintain the current tests.

An `AaveLinearPool` will be added in a later PR.

### CODE REVIEW

**Comment by TomAFrench** [comment]
As we've removed a couple of arguments from the `LinearPool` constructor we can get away without having a struct to avoid the stack issues.

File: pkg/pool-linear/test/LinearPool.test.ts [hunk 1]
```diff
@@ -662,222 +658,4 @@ describe('LinearPool', function () {
...
-
-      context('with a price rate below 1', () => {
-        sharedBeforeEach('mock rate', async () => {
-          await wrappedTokenRateProvider.mockRate(fp(0.99));
-          await pool.updateWrappedTokenRateCache();
-        });
-
-        itAdaptsTheScalingFactorsCorrectly();
-      });
-    });
```
Comment (nventuro) [inline pkg/pool-linear/test/LinearPool.test.ts:735]: These tests are important: they test the rate is correctly applied to the scaling factors. Let's add a simple setter to the `MockLinearPool` for the rate, and keep these tests.

File: pkg/pool-linear/test/LinearPoolFactory.test.ts [hunk 2]
```diff
@@ -22,19 +22,17 @@ describe('LinearPoolFactory', function () {
...
-  const WRAPPED_TOKEN_RATE_CACHE_DURATION = MONTH;
 
   before('setup signers', async () => {
     [, owner] = await ethers.getSigners();
   });
 
   sharedBeforeEach('deploy factory & tokens', async () => {
     vault = await Vault.create();
-    factory = await deploy('LinearPoolFactory', { args: [vault.address] });
+    factory = await deploy('MockLinearPoolFactory', { args: [vault.address] });
```
Comment (nventuro) [inline pkg/pool-linear/test/LinearPoolFactory.test.ts:32]: This is fine as a temporary measure to have these tests pass, but in the future we'll want to replace this back with the AaveLinearPoolFactory.

**Review by nventuro — COMMENTED** [review]
Looks great, love the separation of concerns. Thanks a lot for your help @TomAFrench!

**Review by nventuro — APPROVED** [review]

**Pull request merged**